Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -2489,4 +2489,14 @@ void RenderElement::layoutIfNeeded()
     layout();
 }
 
+SingleThreadWeakPtr<RenderInline> RenderElement::writingSuggestionsRenderer() const
+{
+    return hasRareData() ? rareData().writingSuggestionsRenderer : nullptr;
+}
```

Comment:
Again, is this really a per-renderer rather than a one-per-tree thing?

Reply:
this is a one-per-tree thing; should this RenderInline member also be moved to Editor? or should it stay here?

Reply:
It would be better if we didn't use the renderer to hold on to editing related temporary state(s) if possible.

Reply:
I need to store the renderer somewhere though so that I can update it later on though, right?

Reply:
(similar to backdropRenderer)

Reply:
the renderer is part of the tree, and you can surely hold a (weak) reference to somewhere else (in Editor etc).